### PR TITLE
Project settings

### PR DESCRIPTION
**Pull request by stianjensen**

### CODE REVIEW

File: nin/dasBoot/BEATBEAN.js [hunk 1]
```diff
@@ -5,8 +5,9 @@
     BEAN = 0;
     BEAT = false;
 
-    var subdivision = 6;
-    var beatsPerMinute = 110 * subdivision;
+    var musicInfo = PROJECT.get('music', {});
+    var subdivision = musicInfo.subdivision || 6;
+    var beatsPerMinute = (musicInfo.bpm || 110) * subdivision;
```
Comment (iver56) [inline nin/dasBoot/BEATBEAN.js]: Slightly inaccurate variable name. BPM (Beats Per Minute) is already the number that is set to 110 in this example. The variable holding the number 110 * 6 (in this example) should be called something different.
How about `subBeatsPerMinute` instead?

Comment (sigvef) [inline nin/dasBoot/BEATBEAN.js]: it is more BEATs per minute, since it is the number of beatbean BEATs, e.g. the thing counted by BEAN.

**Comment by sigvef** [comment]
With this strategy, nin.json can never contain anything secret. Is that ok?

**Comment by stianjensen** [comment]
> With this strategy, nin.json can never contain anything secret. Is that ok?

Interesting, I haven't really considered what other things might be stored in this file. Or that a demo project might be secret, for that matter.

Can you come up with something that one could store in nin.json which could be a bad thing to have end up in the compiled version?

**Comment by sigvef** [comment]
> Can you come up with something that one could store in nin.json which could be a bad thing to have end up in the compiled version?

Credentials for some service that we use in a nin build step, for instance.

File: nin/dasBoot/BEATBEAN.js (hunk 1 above)
Comment (aleksanb) [inline nin/dasBoot/BEATBEAN.js]: Agree with iver here. BPM has a well-established meaning in things like FL!

**Comment by aleksanb** [comment]
That's probably a concern we can ignore for quite a while (secrets in nin.json!)

File: nin/dasBoot/BEATBEAN.js (hunk 1 above)
Comment (sigvef) [inline nin/dasBoot/BEATBEAN.js]: I was also agreeing with iver, if that wasn't clear.

**Comment by stianjensen** [comment]
ping

Renamed variables in beatbean.
Also, now only keys which have a default value in projectSettings in the backend will be compiled into the demo. This also gives the guarantee that all PROJECT keys will have a value which makes the API for reading the values nicer.

File: nin/dasBoot/BEATBEAN.js [hunk 2]
```diff
@@ -5,9 +5,8 @@
     BEAN = 0;
     BEAT = false;
 
-    var subdivision = 6;
-    var beatsPerMinute = 110 * subdivision;
-    var beatsPerSecond = beatsPerMinute / 60;
+    var BEATsPerMinute = PROJECT.music.bpm * PROJECT.music.subdivision;
+    var BEATsPerSecond = beatsPerMinute / 60;
```
Comment (sigvef) [inline nin/dasBoot/BEATBEAN.js]: s/beatsPerMinute/BEATsPerMinute/

File: nin/dasBoot/BEATBEAN.js [hunk 3]
```diff
@@ -5,9 +5,8 @@
...
     var framesPerSecond = 60;
     framesPerBeat = framesPerSecond / beatsPerSecond;
```
Comment (sigvef) [inline nin/dasBoot/BEATBEAN.js]: s/beatsPerSecond/BEATsPerSecond/

**Comment by sigvef** [comment]
Consider renaming other variables as well, e.g. framesPerBeat to framesPerBEAT

**Comment by stianjensen** [comment]
ping

**Pull request merged**